docs/389ds/FAQ/bdb-deprecation.md

```diff
@@ -0,0 +1,277 @@
+---
+title: "BDB backend deprecation"
+---
+
+# BDB Deprecation and using lmdb
+
+Since 389-ds-base 2.1.0, The 389 ldap directory server supports two kind of underlying database:
+
+- Berkeley Database (bdb)
+
+- Lightning Memory-Mapped Database Manager (lmdb)
+
+Starting from version 3.0.0 New instances are no more created with bdb by default but with libdb.
+
+Newly created instances are still created with bdb by default while libdb is flagged as [deprecated since Fedora 33]([Changes/Libdb deprecated - Fedora Project Wiki](https://fedoraproject.org/wiki/Changes/Libdb_deprecated)), this change is about to create lmdb instances by default.
+
+{% include toc.md %}
+
+## Why deprecating bdb backend ?
+
+Because the underlying library implementing the bdb version used by 389-ds-base is no more supported upstream and it is already deprecated in some OS:
+
+- [Changes/Libdb deprecated - Fedora Project Wiki](https://fedoraproject.org/wiki/Changes/Libdb_deprecated)
+
+## How to configure a new instance with lmdb or bdb ?
+
+There are several ways:
+
+1. Using dscreate --interactive:
+   Select the proper implementation when the following question is asked:
+
+   ```
+   Choose whether mdb or bdb is used.
+   ```
+
+2. Using dscreate --from-file
+   Set the following stanza in the template file:
+
+   ```
+   [slapd]
+   db_lib = mdb (or bdb)
+   ```
+
+3. Using dscreate with an environment variable.
+   Setting NSSLAPD_DB_LIB environment variable to 'bdb' or 'mdb' change the backend implementation chosen by default
+
+Set  db_lib = mdb (or bdb) in [slapd] section of dscreate template file
+
+## Can I mix different implementations ?
+
+Mixing implementaions limitations are:
+
+- The db_lib parameter is stored in the nsslapd-backend-implement attribute of cn=config entry, so it is global to the instance.
+- backup / restore formats are tied to the db implementation so you cannot use a backup to restore an instance with a different implementation. (ldif export/import and initializing through the replication have not this limitation)
+
+| Question                                                                    | Answer |
+| --------------------------------------------------------------------------- | ------ |
+| Can I mix backends with different implementations in an instance ?          | No     |
+| Can I mix instances with different backends implementations in an host      | Yes    |
+| Can I mix replicas with different implementation in a replicated topology ? | Yes    |
+
+## How to migrate between database implementation ?
+
+There are several ways to do that.
+
+### Automatic method
+
+- dsctl *instanceName* dblib bdb2mdb
```

Comment:
```suggestion
- `dsctl *instanceName* dblib bdb2mdb`
```

Reply:
I would prefer 'dbimplementation' or 'dbformat' than 'dblib'

Reply:
@tbordaz , Several point here:
[1] the FAQ refers to already existing command. We could change the name but IMHO it should be in another ticket as it also impact the code
[2] using long name in command is a bit painful for the users
[3] I do not like any of these proposals.  Maybe rather something like dbmigration (with dbmig as short name)

Reply:
Because the administrator will use this command not frequently (likely only one time), it does not worth the effort of changing an already existing command. I am okay keep 'dblib'.